src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
 
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a >= this.b + this.c
+      || this.b >= this.a + this.c
+      || this.c >= this.b + this.a
+      || this.a <= 0
+      || this.b <= 0
+      || this.c <= 0) {
+      throw new Error('The length of triangle is not correct');
```

Comment:
It's better to use array methods instead of `if` statement with multiple conditions.
For triangle here you have 2 conditions: 
1. All sides should be positive. If it's not true, then throw an Error.
`arr.every(side => side > 0)`
2. Triangle is valid (the longest side is bigger than sum of two another)
Find longest side from array (use array methods or `Math.max`). Find sum of all elements from array (`arr.reduce`). Then you can find validation of triangle with: 
`const validTriangle = longestSide < (sumOfSides - longestSide);`
If it's not true, then throw an error.